### PR TITLE
Add subscription model

### PR DESCRIPTION
**Pull request by serene**
This redesign is intended to hide the complexities of the exact target API. It allows anyone using the gem to use the Subscriber model as any active record model. Saving the subscriber model will just push the changes as intended.

### CODE REVIEW

File: exact_target.gemspec [hunk 1]
```diff
@@ -29,6 +29,9 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency "factory_girl"
   spec.add_dependency "rest-client"
   spec.add_dependency "activesupport"
+  spec.add_dependency 'activerecord'
   spec.add_dependency "builder"
-
-end
\ No newline at end of file
+  spec.add_dependency 'webmock'
```
Comment (rubene) [inline exact_target.gemspec]: maybe test gems should be development dependency only

File: lib/exact_target/subscriber.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+require 'active_record'
```
Comment (rubene) [inline lib/exact_target/subscriber.rb]: you can add this requires to the exact_target.rb

File: lib/exact_target/subscription_manager.rb [hunk 3]
```diff
@@ -0,0 +1,68 @@
+require 'crack'
```
Comment (rubene) [inline lib/exact_target/subscription_manager.rb]: you can add this requires to the exact_target.rb

File: lib/exact_target/subscription_manager.rb [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+require 'hashie'
+
+module ExactTarget
+  class SubscriptionManager
+    def find(email)
+      response = crackie_hash(ExactTarget::Api.new.find_subscriber({ email: email }))
+      build_subscription(email, response.exacttarget.system.subscriber)
+    end
+
+    def exists?(email)
```
Comment (rubene) [inline lib/exact_target/subscription_manager.rb]: can this method be private?

File: lib/exact_target/version.rb [hunk 5]
```diff
@@ -1,3 +1,3 @@
 module ExactTarget
-  VERSION = "0.0.1"
+  VERSION = '0.1.0'
```
Comment (rubene) [inline lib/exact_target/version.rb]: because the functionality will change can we bump to 1.0.0

**Comment by rubene** [comment]
:hankey:

**Comment by serene** [comment]
:wave:

**Comment by rubene** [comment]
:elephant: :elephant: :elephant:

**Pull request merged**